src/test/unit/algorithms/AnomalyTest.cpp

```diff
@@ -36,42 +36,34 @@ using namespace nupic;
 using namespace nupic::sdr;
 
 TEST(ComputeRawAnomalyScore, NoActiveOrPredicted) {
-  SDR active({10});
-  SDR predicted({10});
+  SDR active({10}, {});
+  SDR predicted({10}, {});
   ASSERT_FLOAT_EQ(computeRawAnomalyScore(active, predicted), 0.0);
 };
 
 TEST(ComputeRawAnomalyScore, NoActive) {
-  SDR active({10});
-  SDR predicted({10});
-  predicted.setSparse(SDR_sparse_t{3,5});
-
+  SDR active({10}, {});
+  SDR predicted({10}, {3,5});
   ASSERT_FLOAT_EQ(computeRawAnomalyScore(active, predicted), 0.0f);
 };
 
 TEST(ComputeRawAnomalyScore, PerfectMatch) {
-  SDR active({10});
-  SDR predicted({10});
-  active.setSparse(SDR_sparse_t{3,5,7});
-  predicted.setSparse(SDR_sparse_t{3,5,7});
+  SDR active({10}, {3,5,7});
+  SDR predicted({10}, {3,5,7});
 
   ASSERT_FLOAT_EQ(computeRawAnomalyScore(active, predicted), 0.0f);
 };
 
 TEST(ComputeRawAnomalyScore, NoMatch) {
-  SDR active({10});
-  SDR predicted({10});
-  active.setSparse(SDR_sparse_t{2,4,6});
-  predicted.setSparse(SDR_sparse_t{3,5,7});
+  SDR active({10}, {2,4,6});
```

Comment:
in some cases, the new format is convenient to use. 
Esp as it allows creation of const SDRs //...well, we used mutable, right? Can mutable now be removed and use true const/non-const SDRs?

Reply:
It still needs `mutable`, in order to convert to the other formats after the SDR has been made constant.
```
const SDR A( {1000}, {1,2,3,4} );  <- initialized with sparse format data.
A.getDense();    <- Conversion from sparse to dense happens here.
```